### PR TITLE
CommitHW24

### PR DESCRIPTION
**Pull request by ArturH1990**
CommitHW24 
I know that is wrong, I need help

### CODE REVIEW

File: src/test/resources/Regression.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="Regression" parallel="methods" thread-count="2">
+    <parameter name="BaseURL" value="https://qa.koel.app/"></parameter>
+    <test name="Example Test" preserve-order="false"/>
+
+    <test name="Example Test" preserve-order="false">
```
Comment (rajniss21) [inline src/test/resources/Regression.xml:8]: Two tests in the same suite [Regression] cannot have the same name: Example Test
Remove one test tag.

File: src/test/java/BaseTest.java [hunk 2]
```diff
@@ -1,53 +1,96 @@
...
+        caps.setCapability("browserName","firefox");
+        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
+    case "grid-chrome":
+        caps.setCapability("browserName", "chrome");
+        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
+    default:
+    WebDriverManager.chromedriver().setup();
+    ChromeOptions chromeOptions = new ChromeOptions();
+    chromeOptions.addArguments("--remote-allow-origins=*");
+    return driver = new ChromeDriver();
```
Comment (rajniss21) [inline src/test/java/BaseTest.java:88]: `return driver = new ChromeDriver(chromeOptions);`

File: src/test/java/BaseTest.java [hunk 3]
```diff
@@ -1,53 +1,96 @@
...
+   public void launchBrowser(String BaseURL) throws MalformedURLException, MalformedInputException {
+
 
+//       ChromeOptions options = new ChromeOptions();
+//       options.addArguments("--remote-allow-origins=*")
+//      driver = pickBrowser(System.getProperty("browser"));
 
-       ChromeOptions options = new ChromeOptions();
-       options.addArguments("--remote-allow-origins=*");
+        threadDriver.set(pickBrowser((System.getProperty("browser"))) );
```
Comment (rajniss21) [inline src/test/java/BaseTest.java:49]: `threadDriver.set(pickBrowser((System.getProperty("browser", "chrome"))) );`

File: src/test/java/homework24.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                caps.setCapability("browserName","firefox");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
+            case "grid-chrome":
+                caps.setCapability("browserName", "chrome");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
+            default:
+                WebDriverManager.chromedriver().setup();
+                ChromeOptions chromeOptions = new ChromeOptions();
+                chromeOptions.addArguments("--remote-allow-origins=*");
+                return driver = new ChromeDriver();
```
Comment (rajniss21) [inline src/test/java/homework24.java:33]: Define this whole section in the BaseTest Class and write some tests in homework 24. May be loginTest, deletePlaylistTest, etc.

Comment (ArturH1990) [reply src/test/java/homework24.java:33]: thank you for the correction, may you show an example, please?

Comment (rajniss21) [reply src/test/java/homework24.java:33]: Screenshot_41
Screenshot_40
Screenshot_39

**Comment by rajniss21** [comment]
Hello @ArturH1990 ,
I have checked your homework 24, and I wanted to let you know that you're doing great! Keep up the excellent work!

But you have got this issue while the code has complied. Please review your code, fix it, and complete your homework.
Best wishes from the HW team (Rajnish, rajnish@testpro.io)!
Screenshot_37
Screenshot_38

File: src/test/java/homework24.java (hunk 4 above)
Comment (rajniss21) [reply src/test/java/homework24.java:33]: The BaseTest code looks this this:
`public class BaseTest {
    public static WebDriver driver = null;
    public static WebDriverWait wait = null;
    public static String url = null;
    public static Actions actions = null;

    public static final ThreadLocal<WebDriver> threadDriver = new ThreadLocal<WebDriver>();

    public static WebDriver pickBrowser(String browser) throws MalformedInputException, MalformedURLException {
        DesiredCapabilities caps = new DesiredCapabilities();
        String gridURL = "http://192.168.1.160:4444";

        switch (browser){
            case "firefox":
                WebDriverManager.firefoxdriver().setup();
                return driver = new FirefoxDriver();
            case "grid-firefox":
                caps.setCapability("browserName","firefox");
                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
            case "grid-chrome":
                caps.setCapability("browserName", "chrome");
                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(),caps);
            default:
                WebDriverManager.chromedriver().setup();
                ChromeOptions chromeOptions = new ChromeOptions();
                chromeOptions.addArguments("--remote-allow-origins=*");
                return driver = new ChromeDriver(chromeOptions);
        }
    }


    @DataProvider(name="IncorrectLoginData")
    public Object[][] getDataFromDataProviders() {
        return new Object[][]{
                {"invalid@mail.com", "invalidPass"},
                {"demo@class.com", ""},
                {"", ""}
        };
    }
    @BeforeSuite
    static void setupClass(){
       // WebDriverManager.chromedriver().setup();
    }
    @BeforeMethod
    @Parameters({"BaseURL"})
   public void launchBrowser(String BaseURL) throws MalformedURLException, MalformedInputException {

        threadDriver.set(pickBrowser((System.getProperty("browser", "chrome"))) );
        getDriver().manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        getDriver().manage().window().maximize();

        wait = new WebDriverWait(getDriver(), Duration.ofSeconds(10));
        actions = new Actions(getDriver());//added
        url = BaseURL;
  
        navigateToPage();

    }`

**Pull request merged**